app/model/Google/Repositories/IGoogleRepository.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Model\Mail\Repositories;
+
+use Google_Service_Gmail;
+use Model\Common\UnitId;
+use Model\Google\Exception\OAuthNotFound;
+use Model\Google\OAuth;
+use Model\Google\OAuthId;
+
+interface IGoogleRepository
+{
+    public function getAuthUrl() : string;
+
+    public function saveAuthCode(string $code, UnitId $unitId) : void;
+
+    /** @throws OAuthNotFound */
+    public function find(OAuthId $oauthId) : ?OAuth;
+
+    /**
+     * @param int[] $unitIds
+     *
+     * @return array<int, OAuth[]> unitId => OAuth[]
+     */
+    public function findByUnits(array $unitIds) : array;
+
+    /** @return OAuth[] */
+    public function findByUnit(UnitId $unitId) : array;
+
+    public function remove(OAuth $oAuth) : void;
+
+    public function getGmailService(OAuth $oAuth) : Google_Service_Gmail;
```

Comment:
Tohle nepatří do repository. Mělo by to být (pokud jsem nepřehlédl další use case) součástí `OAuthMailer`u

Reply:
to bych pak musel predavat `Google_Client` mimo repository a to jsem nechtěl

Reply:
To ale v repository nemá co dělat. Jaký je důvod nevytvářet toho klienta přímo v tom maileru?

Repository má jednu responsibility - spravuje persistenci (ukládání a lookup agregátů `OAuth`), žádná další logika (`saveAuthCode()`), ani továrny pro nesouvisející část (samotné odeslání mailu přes výsledek `getGmailService()`) tam nepatří.

Reply:
ok, upravil jsem to a je tam jen getClient

Reply:
Ani getClient() tam nemá být. Ukládání a načítání instancí `OAuth` nemá žádnou souvislost s komunikací s Google API.

Reply:
presunuto do GoogleService